akka-samples/akka-sample-cluster-scala/src/main/scala/sample/cluster/factorial/FactorialFrontend.scala

```diff
@@ -61,10 +61,18 @@ object FactorialFrontend {
     //#registerOnRemoved
     Cluster(system).registerOnMemberRemoved {
       // exit JVM when ActorSystem has been terminated
-      system.registerOnTermination(System.exit(-1))
+      system.registerOnTermination(System.exit(0))
       // in case ActorSystem shutdown takes longer than 10 seconds,
-      // exit the JVM forcefully anyway
-      system.scheduler.scheduleOnce(10.seconds)(System.exit(-1))(system.dispatcher)
+      // exit the JVM forcefully anyway.
+      // Use a Deadline since the scheduled task will also be run when
+      // the ActorSystem is terminated normally.
+      val shutdownDeadline = Deadline.now + 10.seconds
+      system.scheduler.scheduleOnce(shutdownDeadline.timeLeft + 10.millis) {
+        if (shutdownDeadline.isOverdue())
```

Comment:
Ah, this is because the scheduler runs pending events on shutdown? Would it be a heresy to use a normal JVM timer here?